### PR TITLE
Mpu6050 Accel/Gyro

### PR DESCRIPTION
**Pull request by vsyn**

### CODE REVIEW

File: hw/sensor/creator/src/sensor_creator.c [hunk 1]
```diff
@@ -93,6 +100,14 @@ static struct sensor_itf i2c_0_itf_lsm = {
 };
 #endif
 
+#if MYNEWT_VAL(I2C_3) && MYNEWT_VAL(MPU6050_OFB)
```
Comment (vrahane) [inline hw/sensor/creator/src/sensor_creator.c]: For the sensor_creator, we would like to keep everything generic and hence I2C_0 is used, I would keep it in sync with everything and change I2C_3 to I2C_0.

File: hw/drivers/sensors/mpu6050/src/mpu6050.c [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+ *
+ * @param The sensor interface
+ * @param The I2C address to use
+ * @param The register address to read from
+ * @param Pointer to where the register value should be written
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+mpu6050_read8(struct sensor_itf *itf, uint8_t addr, uint8_t reg,
```
Comment (vrahane) [inline hw/drivers/sensors/mpu6050/src/mpu6050.c]: The device I2C interface address is already present in sensor_itf as "si_addr". It generally gets set in the bsp for onboard sensors and in the creator for offboard ones. The parameter "addr" is not needed. You already set the interface address. All you would have to do here is use the macro "SENSOR_GET_ITF()" to get the interface and then use "si_addr" field.

File: hw/drivers/sensors/mpu6050/src/mpu6050.c [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+ *
+ * @param The sensor interface
+ * @param The I2C address to use
+ * @param The register address to write to
+ * @param The value to write
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+mpu6050_write8(struct sensor_itf *itf, uint8_t addr, uint8_t reg,
```
Comment (vrahane) [inline hw/drivers/sensors/mpu6050/src/mpu6050.c]: The device I2C interface address is already present in sensor_itf as "si_addr". It generally gets set in the bsp for onboard sensors and in the creator for offboard ones. The parameter "addr" is not needed. You already set the interface address. All you would have to do here is use the macro "SENSOR_GET_ITF()" to get the interface and then use "si_addr" field.

File: hw/drivers/sensors/mpu6050/src/mpu6050.c [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+ *
+ * @param The sensor interface
+ * @param The I2C address to use
+ * @param The register address to read from
+ * @param Pointer to where the register value should be written
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+mpu6050_read48(struct sensor_itf *itf, uint8_t addr, uint8_t reg,
```
Comment (vrahane) [inline hw/drivers/sensors/mpu6050/src/mpu6050.c]: The device I2C interface address is already present in sensor_itf as "si_addr". It generally gets set in the bsp for onboard sensors and in the creator for offboard ones. The parameter "addr" is not needed. You already set the interface address. All you would have to do here is use the macro "SENSOR_GET_ITF()" to get the interface and then use "si_addr" field.

File: hw/drivers/sensors/mpu6050/include/mpu6050/mpu6050.h [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+#define MPU6050_ADDR_ACCEL                  0x19  /* 0011001 */
+#define MPU6050_ADDR_MAG                    0x1E  /* 0011110 */
+
+struct mpu6050_cfg {
+    enum mpu6050_accel_range accel_range;
+    enum mpu6050_gyro_range gyro_range;
+    uint8_t gyro_rate_div; /* Sample Rate = Gyroscope Output Rate /
+            (1 + gyro_rate_div) */
+    uint8_t lpf_cfg; /* See data sheet */
+    uint8_t addr;
```
Comment (vrahane) [inline hw/drivers/sensors/mpu6050/include/mpu6050/mpu6050.h]: In the cfg you do not need the I2C interface address for the device as it is already present in "si_addr".

File: hw/drivers/sensors/mpu6050/src/mpu6050_priv.h [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+#ifndef __MPU6050_PRIV_H__
+#define __MPU6050_PRIV_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum mpu6050_registers {
+    MPU6050_SELF_TEST_X = 0x0d, // RW
```
Comment (vrahane) [inline hw/drivers/sensors/mpu6050/src/mpu6050_priv.h]: Our coding style does not allow c++ style comments, hence all the "// RW and //R " comments should either be converted to C comments which is "/* RW */ and /* R */ or removed.

**Comment by vsyn** [comment]
Hopefully this addresses your concerns, I am working on a version that supports interrupts with a few other minor changes.

File: hw/drivers/sensors/mpu6050/src/mpu6050.c [hunk 7]
```diff
@@ -140,7 +136,7 @@ mpu6050_read8(struct sensor_itf *itf, uint8_t addr, uint8_t reg,
     rc = hal_i2c_master_write(itf->si_num, &data_struct,
                               OS_TICKS_PER_SEC / 10, 0);
     if (rc != 0) {
-        MPU6050_ERR("I2C access failed at address 0x%02X\n", addr);
+        MPU6050_ERR("I2C access failed at address 0x%02X\n", itf->si_add);
```
Comment (vrahane) [inline hw/drivers/sensors/mpu6050/src/mpu6050.c]: "itf->si_add" should be "itf->si_addr" for all the log lines where address is logged.

**Pull request merged**